lib/utils.ts

```diff
@@ -59,10 +76,20 @@ export function generateBuildTasks(
 				img.image.dockerfile != null
 					? { dockerfilePath: img.image.dockerfile }
 					: {},
-				// It's possible to specify an image name as well as
-				// a build, but that doesn't make sense in a balena
-				// ecosystem, so we remove it
-				_.omit(img.image, 'image'),
+				// Pass through args, context, labels, tag.
+				imageProps,
+				// Pass through build options from composition
+				// translating to dockerode ImageBuildOptions properties
+				{
+					dockerOpts: _.merge(
+						{},
+						cachefrom ? { cachefrom } : {},
+						shmsize ? { shmsize } : {},
+						target ? { target } : {},
+						extrahosts ? { extrahosts } : {},
+					),
```

Comment:
> `extrahosts` _string_ Extra hosts to add to /etc/hosts
> https://docs.docker.com/engine/api/v1.41/#operation/ImageBuild

What a peculiar image build option! 😀 

As these options don't overlap with each other, I believe JS's builtin spread syntax would be sufficient as follows, but it does not really matter:
```suggestion
					dockerOpts: {
						...(cachefrom ? { cachefrom } : {}),
						...(shmsize ? { shmsize } : {}),
						...(target ? { target } : {}),
						...(extrahosts ? { extrahosts } : {}),
					},
```